src/d3a/models/strategy/storage.py

```diff
@@ -126,6 +126,22 @@ def area_reconfigure_event(self, risk=None, initial_rate_option=None,
             self.state.min_allowed_soc = min_allowed_soc
 
     def event_activate(self):
+        if ConstSettings.IAASettings.PRICING_SCHEME != 0:
```

Comment:
Since this is a repetitive pattern to assign these on the event_activate method, I would create a base class method on the BaseStrategy that would be overridden by the subclasses with these conditions. That way there will be a common entry point for all strategies, and it might also be possible to use the BaseStrategy event_activate method for calling this entry point. That way the event_activate methods of each individual strategy will remain untouched. If it is not that clear let me know to discuss this offline.

Reply:
I know what you mean, but here this pattern is used to set the break even point, which is storage-specific. 
Additionally, only in the storage this pattern is called in the event_activate. 
For setting the sell prices in the PV and Storage I already refactored this pattern into set_initial_selling_rate_alternative_pricing_scheme() . 

Or do you mean only the following two lines?:
```
        if ConstSettings.IAASettings.AlternativePricing.PRICING_SCHEME != 0:
            self.assign_offermixin_arguments(3, 2, 0)
```

Reply:
For the latter, I could refactor it into the base class.

Reply:
Spyros and me did offline discussion and refactoring.